src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SyntaxTokenExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using SyntaxKind = Microsoft.AspNetCore.Razor.Language.SyntaxKind;
+using SyntaxToken = Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxToken;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal static class SyntaxTokenExtensions
+    {
+        public static bool IsWhitespace(this SyntaxToken token)
+        {
+            return token.Kind == SyntaxKind.Whitespace || token.Kind == SyntaxKind.NewLine;
+        }
+
+        public static bool IsSpace(this SyntaxToken token)
+        {
+            return token.Kind == SyntaxKind.Whitespace && token.Content.Equals(" ", StringComparison.Ordinal);
```

Comment:
Ugh, I hate that we have to do this. I wonder if C# has a singular Whitespace syntax kind or if they have spaces and tabs etc.